onnxruntime/core/providers/cpu/tensor/cast_op.cc

```diff
@@ -254,11 +254,16 @@ struct TensorCasterNoSat<std::string, DstType> {
 // tensor MLFloat16 -> float
 template <>
 struct TensorCaster<MLFloat16, float> {
-  void Cast(const OpKernelContext&, const TensorShape& shape, const Tensor& in, Tensor& out) const {
+  void Cast(const OpKernelContext& ctx, const TensorShape& shape, const Tensor& in, Tensor& out) const {
     auto out_data = out.MutableData<float>();
     auto in_data = in.Data<MLFloat16>();
     const size_t shape_size = narrow<size_t>(shape.Size());
-    MlasConvertHalfToFloatBuffer(in_data, out_data, shape_size);
+
+    concurrency::ThreadPool::TryParallelFor(ctx.GetOperatorThreadPool(), shape_size,
+                                            {shape_size * 2.f, shape_size * 4.f, static_cast<double>(shape_size / 4)},
```

Comment:
The PR needs benchmarking for different sizes.
We do not want to spawn many threads for small tensors.

Reply:
Did some testing on LNL and got these results, adapted the code so that threads are used only on tensors >128k
[mt_v_main shape.csv](https://github.com/user-attachments/files/19198351/mt_v_main.shape.csv)